File: review-agent-server/src/main/java/com/review/agent/domain/entity/Review.java
package com.review.agent.domain.entity;

import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;
import com.review.agent.domain.enums.ReviewMode;
import com.review.agent.domain.enums.ReviewStatus;
import lombok.Data;

import java.time.LocalDateTime;

/**
 * 代码审查任务实体
 */
@Data
@TableName("review")
public class Review {

    @TableId(type = IdType.AUTO)
    private Long id;

    /** 关联的项目 ID */
    private Long projectId;

    /** 源分支 */
    private String sourceBranch;

    /** 目标分支 */
    private String targetBranch;

    /** 源 commit SHA */
    private String sourceCommit;

    /** 目标 commit SHA */
    private String targetCommit;

    /** 审查状态 */
    private ReviewStatus status;

    /** 审查模式：单模型 / 多模型 / 裁判模式 */
    private ReviewMode reviewMode;

    /** JSON 格式的模型配置 */
    private String modelsConfig;

    /** 审查摘要 JSON */
    private String summary;

    /** diff 原始内容 */
    private String diffContent;

    private LocalDateTime createdAt;
    private LocalDateTime updatedAt;
}

File: review-agent-web/types/review.ts
export type ReviewStatus = 'PENDING' | 'RUNNING' | 'COMPLETED' | 'FAILED'
export type ReviewMode = 'SINGLE' | 'MULTI' | 'JUDGE' | 'AGENT'
export type SeverityLevel = 'BLOCKER' | 'MAJOR' | 'MINOR' | 'INFO'
export type FindingCategory = 'CODE_STYLE' | 'BUG' | 'PERFORMANCE' | 'SECURITY' | 'EXCEPTION_HANDLING' | 'OTHER'
export type HumanStatus = 'PENDING' | 'CONFIRMED' | 'DISMISSED'
export type ModelRole = 'WORKER' | 'JUDGE'

export interface Review {
  id: number
  projectId: number
  projectName: string | null
  sourceBranch: string | null
  targetBranch: string | null
  sourceCommit: string | null
  targetCommit: string | null
  status: ReviewStatus
  reviewMode: ReviewMode
  modelsConfig: string | null
  summary: string | null
  createdAt: string
  updatedAt: string
}

export interface ReviewFinding {
  id: number
  reviewId: number
  filePath: string
  lineStart: number | null
  lineEnd: number | null
  category: FindingCategory
  severity: SeverityLevel
  title: string
  description: string | null
  suggestion: string | null
  modelName: string | null
  confidence: number | null
  isCrossHit: boolean
  humanStatus: HumanStatus
  createdAt: string
}

export interface ReviewModelResult {
  id: number
  reviewId: number
  modelName: string
  role: ModelRole
  status: ReviewStatus
  rawResult: string | null
  errorMessage: string | null
  startedAt: string | null
  completedAt: string | null
}

export interface ReviewDetail extends Review {
  findings: ReviewFinding[]
  modelResults: ReviewModelResult[]
  totalFindings: number
  blockerCount: number
  majorCount: number
  minorCount: number
  infoCount: number
  prePrStatus: string | null
  blockedReasons: string[] | null
}

export interface CreateReviewParams {
  projectId: number
  sourceBranch?: string
  targetBranch?: string
  sourceCommit?: string
  targetCommit?: string
  reviewMode: ReviewMode
  modelsConfig?: string
}

export interface CreatePrePrParams {
  projectId: number
  sourceBranch: string
  targetBranch: string
}

export type AgentRole = 'SECURITY_AUDITOR' | 'PERFORMANCE_ANALYST' | 'CODE_STYLE_CHECKER' | 'EXCEPTION_HANDLER' | 'ARCHITECT_REVIEWER'

export interface AgentConfig {
  role: AgentRole
  modelName: string
  systemPrompt?: string
  skills: string[]
  temperature?: number
  maxSteps?: number
}

export interface AgentPipelineConfig {
  agents: AgentConfig[]
  orchestrationStrategy: 'PARALLEL' | 'SEQUENTIAL' | 'HYBRID'
  mcpEnabled: boolean
  mcpServerUrls: string[]
}

export interface AgentExecutionResult {
  role: string
  modelName: string
  success: boolean
  output: string | null
  errorMessage: string | null
  durationMs: number
}

export interface AgentRoleOption {
  role: AgentRole
  label: string
  defaultModel: string
  skills: string[]
  description: string
}

export interface ModelOption {
  value: string
  label: string
}

export interface AgentReviewDetail extends ReviewDetail {
  agentResults: AgentExecutionResult[]
}
